clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -130,17 +389,136 @@ mlir::Value ScalarExprEmitter::VisitCastExpr(CastExpr *ce) {
   QualType destTy = ce->getType();
   CastKind kind = ce->getCastKind();
 
+  // These cases are generally not written to ignore the result of evaluating
+  // their sub-expressions, so we clear this now.
+  [[maybe_unused]] bool ignored = TestAndClearIgnoreResultAssign();
```

Comment:
```suggestion
  TestAndClearIgnoreResultAssign();
```

More interestingly, why not just ignoreResultAssign = false; or something?